src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,345 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.reporting
+
+import org.apache.commons.io.FileUtils
+import org.bitbucket.inkytonik.kiama.relation.NodeNotInTreeException
+import viper.gobra.ast.frontend.{PDomainFunction, PExpression, PFPredicateDecl, PFunctionDecl, PMPredicateDecl, PMPredicateSig, PMethodDecl, PMethodImplementationProof, PMethodSig, PNode, PParameter, PPredConstructor}
+import viper.gobra.ast.internal.BuiltInMember
+import viper.gobra.frontend.{Config, PackageEntry}
+import viper.gobra.frontend.info.{Info, TypeInfo}
+import viper.gobra.util.Violation
+import viper.silver.ast.{FuncApp, Function, Member, Method, MethodCall, Node, Predicate, PredicateAccess}
+import viper.silver.reporter.Time
+
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.annotation.tailrec
+import java.io.File
+
+case class StatsCollector(reporter: GobraReporter) extends GobraReporter {
+  // Stores type info for a task
+  var typeInfos: Map[String, TypeInfo] = Map()
+  // Maps a gobra member name to a gobra member entry
+  private var memberMap: Map[String, GobraMemberEntry] = Map()
+  // Maps a viper member name to a gobra member entry
+  private var viperMemberNameGobraMemberMap: Map[String, GobraMemberEntry] = Map()
+
+  case class GobraMemberEntry(pkgDir: String,
+                              pkg: String,
+                              memberName: String,
+                              nodeType: String,
+                              args: String,
+                              var viperMembers: Map[String, ViperMemberEntry],
```

Comment:
Is there an easy way to avoid having mutable fields here? Couldn't we move this map to a private field of the `StatsCollector`?